### ISSUE TITLE
Add attributes that apply the "standard" JsonConverters

### ISSUE THREAD

**Issue by jennings** [nodatime/nodatime.serialization#91]
System.Text.Json provides two ways to specify the JsonConverter used to convert a type:

1. Add the converter to the `JsonSerializerOptions.Converters` collection
2. Add a `JsonConverterAttribute` to the property to be serialized

Option 1 works great, but it requires constructing or passing around a properly-configured JsonSerializerOptions. In our apps, there is no situation where I want an `Instant` property to be serialized using anything except `NodaConverters.InstantConverter`.

It would be convenient to provide a `JsonConverterAttribute` that can be used instead of configuring a `JsonSerializerOptions`. However, NodaTime.Serialization.SystemTextJson can't do this out of the box because the converters are static properties on `NodaConverters`, and so can't be passed to the `JsonConverterAttribute` constructor (which takes a `Type`).

One option is a single attribute that can be applied to almost any NodaTime type:

```csharp
public class NodaTimeDefaultJsonConverterAttribute : JsonConverterAttribute
{
    public override JsonConverter? CreateConverter(Type typeToConvert)
    {
        if (typeToConvert == typeof(Instant))
        {
            return NodaConverters.InstantConverter;
        }
        else if (typeToConvert == typeof(Interval))
        {
            return NodaConverters.IntervalConverter;
        }
        // ...
        else
        {
            return null;
        }
    }
}

// Usage:
public class A
{
    [NodaTimeDefaultJsonConverter]
    public Instant CreatedAt { get; set; }

    [NodaTimeDefaultJsonConverter]
    public LocalDate Birthday { get; set; }
}
```

This couldn't be applied to properties of type `DateTimeZone` or `ZonedDateTime`, since a `IDateTimeZoneProvider` needs to be specified. To avoid making a special case for those two types, a second option would be to provide a separate `JsonConverterAttribute` for each NodaTime type:

```csharp
public class InstantConverterAttribute : JsonConverterAttribute
{
    JsonConverter? _converter;

    public InstantConverterAttribute() =>
        _converter = NodaConverters.InstantConverter;

    public override JsonConverter? CreateConverter(Type typeToConvert)
        => _converter;
}

public class TzdbDateTimeZoneConverterAttribute : JsonConverterAttribute
{
    JsonConverter? _converter;

    public TzdbDateTimeZoneConverterAttribute() =>
        _converter = NodaConverters.CreateDateTimeZoneConverter(DateTimeZoneProviders.Tzdb);

    public override JsonConverter? CreateConverter(Type typeToConvert)
        => _converter;
}

// Usage:
public class B
{
    [InstantConverter]
    public Instant CreatedAt { get; set; }

    [LocalDateConverter]
    public LocalDate Birthday { get; set; }

    [TzdbDateTimeZoneConverter]
    public DateTimeZone DisplayTimeZone { get; set; }
}
```

The converters could be written to allow custom patterns, if desired:

```csharp
[InstantConverter("uuuu-MM-dd HH:mm:ss")] // would construct a NodaPatternConverter<Instant> on first use
```

**Comment by jskeet** [comment]
Thanks for the detailed feature request. I really need to get back to the handful of serialization issues we've got building up (with half-completed PRs). I'll *hope* to get to this in a few weeks, but it probably wouldn't be before then. I do like the idea of being able to apply it with a custom pattern...

**Comment by jennings** [comment]
Are you open to contributions on this? I can't guarantee I'll have time either, but maybe I'll need something straightforward I can build as a palate cleanser.

**Comment by jskeet** [comment]
Yes, absolutely open to contributions. That would be great :)

**Comment by jskeet** [comment]
FYI, I'm *hoping* to get to this today - basically because the issue was so well written, it goes to the top of the pile :)

**Comment by jskeet** [comment]
Fixed in #94.
